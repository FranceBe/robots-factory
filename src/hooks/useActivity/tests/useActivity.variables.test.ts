import {
  emptyInfo,
  infoByActivity,
  nameByActivity,
  timeBaseByActivity,
} from 'hooks/useActivity/useActivity.variables'

describe('useActivity variables', () => {
  describe('emptyInfo', () => {
    it('should match snapshot', () => {
      expect(emptyInfo).toMatchSnapshot()
    })
  })
  describe('infoByActivity', () => {
    it('should match snapshot', () => {
      expect(infoByActivity).toMatchSnapshot()
    })
  })
  describe('nameByActivity', () => {
    it('should match snapshot', () => {
      expect(nameByActivity).toMatchSnapshot()
    })
  })
  describe('timeBaseByActivity', () => {
    it('should match snapshot', () => {
      expect(timeBaseByActivity).toMatchSnapshot()
    })
  })
})
