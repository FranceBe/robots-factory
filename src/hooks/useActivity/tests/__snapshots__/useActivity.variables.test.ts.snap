// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useActivity variables emptyInfo should match snapshot 1`] = `
Object {
  "iconInfo": undefined,
  "textInfo": "",
}
`;

exports[`useActivity variables infoByActivity should match snapshot 1`] = `
Object {
  "bar": Object {
    "current": Object {
      "iconInfo": Object {
        "spin": true,
        "type": "spinner",
      },
      "textInfo": "Le robot est en train de miner du Bar. Cela prend entre 0.5 et 2s / Bar.",
    },
    "failure": Object {
      "iconInfo": undefined,
      "textInfo": "",
    },
    "success": Object {
      "iconInfo": Object {
        "color": "#45B69C",
        "type": "success",
      },
      "textInfo": "Le robot a fini d'assembler un Bar.",
    },
  },
  "foo": Object {
    "current": Object {
      "iconInfo": Object {
        "spin": true,
        "type": "spinner",
      },
      "textInfo": "Le robot est en train de miner du Foo. Cela prend 1s / Foo.",
    },
    "failure": Object {
      "iconInfo": undefined,
      "textInfo": "",
    },
    "success": Object {
      "iconInfo": Object {
        "color": "#45B69C",
        "type": "success",
      },
      "textInfo": "Le robot a fini d'assembler un Foo.",
    },
  },
  "foobar": Object {
    "current": Object {
      "iconInfo": Object {
        "spin": true,
        "type": "spinner",
      },
      "textInfo": "Le robot est en train d'assembler un Foobar. Cela prend 2s et a 60% de chance de succès.",
    },
    "failure": Object {
      "iconInfo": Object {
        "color": "#F45B69",
        "type": "failure",
      },
      "textInfo": "Echec : Le robot n'a pas assemblé de Foobar, le Bar a pu être sauvé mais le Foo est perdu.",
    },
    "success": Object {
      "iconInfo": Object {
        "color": "#45B69C",
        "type": "success",
      },
      "textInfo": "Succès : Le robot a fini d'assembler un Foobar.",
    },
  },
  "moving": Object {
    "current": Object {
      "iconInfo": Object {
        "type": "moving",
      },
      "textInfo": "Le robot est en train de changer d'activité. Cela prend 5s.",
    },
    "failure": Object {
      "iconInfo": undefined,
      "textInfo": "",
    },
    "success": Object {
      "iconInfo": undefined,
      "textInfo": "",
    },
  },
  "robot": Object {
    "current": Object {
      "iconInfo": undefined,
      "textInfo": "",
    },
    "failure": Object {
      "iconInfo": undefined,
      "textInfo": "",
    },
    "success": Object {
      "iconInfo": Object {
        "color": "#45B69C",
        "type": "success",
      },
      "textInfo": "Succès : Le robot a acheté un nouveau robot.",
    },
  },
}
`;

exports[`useActivity variables nameByActivity should match snapshot 1`] = `
Object {
  "bar": "bar",
  "foo": "foo",
  "foobar": "foobar",
  "moving": "moving",
  "robot": "robot",
}
`;

exports[`useActivity variables timeBaseByActivity should match snapshot 1`] = `
Object {
  "foo": 1,
  "foobar": 2,
  "moving": 5,
  "robot": 0,
}
`;
